library/std/src/sys/unix/thread.rs

```diff
@@ -113,7 +113,7 @@ impl Thread {
 
     pub fn yield_now() {
         let ret = unsafe { libc::sched_yield() };
-        debug_assert_eq!(ret, 0);
+        assert_eq!(ret, 0, "unexpected error during sched_yield: {:?}", ret);
```

Comment:
I pointed out here that it might be better to just ignore this error: https://github.com/rust-lang/rust/issues/94705#issuecomment-1062086068